### ISSUE TITLE
2.0.0-alpha-2: Inheritance Method problem

### ISSUE THREAD

**Issue by KaiHufenbach** [gaearon/react-hot-loader#184]
After updating to the latest version I am experiencing serious issues on method search:

A.js:
```
class A {

  render(){
    console.log(this.getContent());
  }
}

export default A;
```

B.js:
```
import A from './A';

class B extends A {
  getContent(){
    return 'hooray';
  }
}

export default B;
```

now trying to invoke:

```
let b = new B();
b.render();
```

throws:
![image](https://cloud.githubusercontent.com/assets/959063/9463592/e9a2176c-4b1d-11e5-9579-4a50c5b66cc2.png)

Even more confusing:

```
...
b.getContent();
```
throws the same exception.

The method is not accessable as long as B extends A. Just remove ... extends A ... then b.getContent can be accessed.

**Comment by gaearon** [comment]
Thanks for reporting! You shouldn't use inheritance for React components, it's generally a bad idea. However you're right we at least shouldn't break it. Please revert to `react-hot-loader@1.2.9` until this issue is solved.

**Comment by gaearon** [comment]
The right way to solve this problem with composition:

```js
class A {
  return <div>{this.props.children}</div>
}

class B {
  return <A>hooray</A>;
}
```

**Comment by gaearon** [comment]
Because this is a React Proxy issue, let's track it in https://github.com/gaearon/react-proxy/issues/9.

**Comment by gaearon** [comment]
Should be fixed in `2.0.0-alpha-3`, please verify.
